react/src/components/Chat/ChatCard.tsx

```diff
@@ -83,35 +80,48 @@
 function useModels(
   provider: ChatProviderType,
   fetchKey: string,
-  endpoint?: ChatCard_endpoint$data | null,
+  baseURL?: string,
+  token?: string,
 ) {
+  const { message: appMessage } = App.useApp();
   const { data: modelsResult } = useSuspenseTanQuery<{
     data: Array<Model>;
   }>({
-    queryKey: ['models', fetchKey, endpoint?.endpoint_id],
-    queryFn: () => {
-      return endpoint?.url
-        ? fetch(
-            new URL(
-              provider.basePath + '/models',
-              endpoint?.url ?? undefined,
-            ).toString(),
-          )
-            .then((res) => res.json())
-            .catch((e) => ({ data: [] }))
-        : Promise.resolve({ data: [] });
+    queryKey: ['models', fetchKey, baseURL, token],
+    queryFn: async () => {
+      try {
+        if (baseURL) {
+          const { protocol, host, pathname: path } = new URL(baseURL);
+          const url = new URL(`${path}/models`, protocol + host).toString();
+          const res = await fetch(url, {
+            headers: {
+              Authorization: `Bearer ${token || provider.apiKey || 'dummy'}`,
+              'Content-Type': 'application/json',
+            },
+          });
+
+          if (!res.ok) {
+            throw new ChatModelError(res.status);
+          }
+
+          return await res.json();
+        }
+      } catch (error: any) {
+        appMessage.error(`Error fetching models: ${error.message}`, 5);
```

Comment:
I'm not sure about that showing error message in catch block. Let me know if you have a better way to manage it. I guess we need another state but I'm not sure that we should add more state for error messages